core/Dispatch.cpp

```diff
@@ -31,38 +32,96 @@ namespace olympia
                                                                              info_logger_,
                                                                              in_credit_port.get(),
                                                                              out_inst_port.get()));
+                }
+            }
+        }
+
+        /*
+            We're creating a pipe mapping for each issue queue types:
+            pipe_map_ = {"int": ["issue_queue"]}
+            so we have to find for each pipe, what issue queue is mapped to this, which is a search over
+            pipe_topology and issue_queue_topology
+            For example if we have the topology defined for alus as:
+            pipe_topology_alu_pipes:
+                [["int"], # alu0
+                ["int", "div"], # alu1
+                ["int", "mul"], # alu2
+                ["int", "mul", "i2f", "cmov"], # alu3
+                ["int"], # alu4
+                ["int"] # alu5]
+            and an issue queue topology of:
+            issue_queue_topology:
+                [["alu0"], # issue queue 0
+                ["alu1", "alu2"], # issue queue 1
+                ["alu3", "alu4", "alu5"], # issue queue 2
+                ["fpu0", "fpu1"],
+                ["br0", "br1"]]
+            The algorithm below loops over the issue_queue_topology, finds the associated pipe topology for it,
+            then looks up the associated issue queue dispatcher to associate with that pipe topology.
+
+            Example flow:
+                Iterate onto alu0, look at pipe_toplology_alu_pipes[0] which is "int", 
+                set in pipe_map_ "int": issue_queue0 dispatcher
+        */
+       auto setup_pipe_map = [this, issue_queue_pipe_topology, node](std::string exe_unit){
+            auto exe_unit_pipe_topology = olympia::coreutils::getPipeTopology(node->getParent(), "pipe_topology_" + exe_unit + "_pipes");
+            std::unordered_map<std::string, std::vector<std::string>> exe_unit_to_pipe_mapping;
+            // Get mapping of execution 
+            for(size_t i = 0; i < exe_unit_pipe_topology.size(); ++i){
+                std::string unit_name = exe_unit + std::to_string(i);
+                exe_unit_to_pipe_mapping[unit_name] = exe_unit_pipe_topology[i];
+            }
+            std::unordered_map<std::string, std::shared_ptr<Dispatcher>> string_to_dispatcher;
+            const auto target_itr = InstArchInfo::dispatch_target_map.find(exe_unit);
+            auto & dispatchers = dispatchers_[target_itr->second];
+            for(auto & dispatcher: dispatchers){
+                string_to_dispatcher[dispatcher->getName()] = dispatcher;
+            }
+
+            for(size_t j = 0; j < issue_queue_pipe_topology.size(); ++j){
+                for(size_t k = 0; k < issue_queue_pipe_topology[j].size(); ++k){
+                    std::string tgt_unit_name = issue_queue_pipe_topology[j][k];
```

Comment:
Can be `const` -- please look for other situations where this can be true.  I'll stop harping on it.  😄

Reply:
Resolved in [6b0e363](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/6b0e3633c1b7331170afda8920f596f9fd16b233)